contracts/src/base/registry/facets/distribution/RewardsDistribution.sol

```diff
@@ -33,65 +33,110 @@ contract RewardsDistribution is
     RewardsDistributionStorage.Layout storage ds = RewardsDistributionStorage
       .layout();
+    return ds.distributionByOperator[operator];
+  }
 
-    uint256 totalClaimableAmount = 0;
+  function getClaimableAmountForDelegator(
+    address delegator
+  ) public view returns (uint256) {
+    RewardsDistributionStorage.Layout storage ds = RewardsDistributionStorage
+      .layout();
+    return ds.distributionByDelegator[delegator];
+  }
 
+  function getClaimableAmountForAuthorizedClaimer(
+    address claimer
+  ) public view returns (uint256) {
+    RewardsDistributionStorage.Layout storage ds = RewardsDistributionStorage
+      .layout();
     address[] memory delegatorsForClaimer = _getDelegatorsByAuthorizedClaimer(
       claimer
     );
+    uint256 totalClaimableAmount = 0;
     for (uint256 i = 0; i < delegatorsForClaimer.length; i++) {
       totalClaimableAmount += ds.distributionByDelegator[
         delegatorsForClaimer[i]
       ];
     }
+    return totalClaimableAmount;
+  }
+
+  function operatorClaim() external {
+    RewardsDistributionStorage.Layout storage ds = RewardsDistributionStorage
+      .layout();
 
     NodeOperatorStorage.Layout storage nos = NodeOperatorStorage.layout();
     address[] memory operatorsForClaimer = nos
-      .operatorsByClaimer[claimer]
+      .operatorsByClaimer[msg.sender]
       .values();
-    for (uint256 i = 0; i < operatorsForClaimer.length; i++) {
-      totalClaimableAmount += ds.distributionByOperator[operatorsForClaimer[i]];
-    }
 
-    totalClaimableAmount += ds.distributionByDelegator[claimer];
+    uint256 amount = 0;
+    for (uint256 i = 0; i < operatorsForClaimer.length; i++) {
+      amount += getClaimableAmountForOperator(operatorsForClaimer[i]);
 
-    return totalClaimableAmount;
-  }
+      ds.distributionByOperator[operatorsForClaimer[i]] = 0;
+    }
 
-  function claim() external {
-    uint256 amount = getClaimableAmount(msg.sender);
     if (amount == 0) revert RewardsDistribution_NoRewardsToClaim();
 
     SpaceDelegationStorage.Layout storage sd = SpaceDelegationStorage.layout();
     if (IERC20(sd.riverToken).balanceOf(address(this)) < amount)
       revert RewardsDistribution_InsufficientRewardBalance();
 
+    CurrencyTransfer.transferCurrency(
+      sd.riverToken,
+      address(this),
+      msg.sender,
+      amount
+    );
+  }
+
+  function mainnetClaim() external {
     RewardsDistributionStorage.Layout storage ds = RewardsDistributionStorage
       .layout();
-
-    //clear all this claimers own rewards
-    ds.distributionByDelegator[msg.sender] = 0;
-    ds.distributionByOperator[msg.sender] = 0;
-
-    //clear all the delegators rewards for this claimer
     address[] memory delegatorsForClaimer = _getDelegatorsByAuthorizedClaimer(
       msg.sender
     );
+    uint256 totalClaimableAmount = 0;
     for (uint256 i = 0; i < delegatorsForClaimer.length; i++) {
+      totalClaimableAmount += ds.distributionByDelegator[
```

Comment:
You can just call `getClaimableAmountForDelegator(delegatorsForClaimer[i])` here and remove line 100 ?